src/services/email.service.ts

```diff
@@ -0,0 +1,42 @@
+import * as SendGridMail from '@sendgrid/mail';
+import * as dotenv from 'dotenv';
+
+dotenv.config();
+SendGridMail.setApiKey(process.env.SENDGRID_API_KEY);
+
+/* Send an email to one or more recipients */
+const _sendEmails = async (
+  emailInfoArr: SendGridMail.MailDataRequired[]
+): Promise<void> => {
+  const [
+    {
```

Comment:
I don't see why we have to destructure here, aren't you extracting and reassembling the fields into the same obj? what might be useful is instead of exposing the SendGridMail as the param to the public method, we create our own mail struct/interface/type or whatever it's called in ts and translate that to sendgridmail.mailxxx

that way we don't break the email abstraction

Reply:
The destructuring was like a way for me to "expose" the fields I guess... And yes I can write a type interface to use instead of having to use SendGridMail.MailDataRequired[] for emailInfoArr.

Reply:
Where should I put this type definition file though? You want me to make a whole new folder for it?

Reply:
I'm also foreseeing the main usecase being sending a single email bcc-ed to a lot of people

Reply:
for now each service can be it's own folder inside services with service + types in two files

Reply:
idr the difference between interfaces and types in ts so you'll have to read up

Reply:
I'll restrict the interface's properties to just what we listed in the docs + 'from", is that okay? Otherwise if we just take all of what SendGridMail.MailDataRequired has then we might as well just use their data type.

Reply:
could perhaps also use a typealias https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-aliases

Reply:
Yep. In our case, it's the same whether we use interface or type alias. There are differences between them but for the things we are doing here we don't need to worry about it just yet.